### PR TITLE
Endorsements ithaca backend

### PR DESCRIPTION
**Pull request by akoptelov**

### CODE REVIEW

File: rpc/src/services/dev_services.rs [hunk 1]
```diff
@@ -910,16 +914,79 @@ pub(crate) async fn get_shell_automaton_endorsing_rights(
...
+    round: Option<i32>,
+    base_time: Option<u64>,
     env: &RpcServiceEnvironment,
 ) -> anyhow::Result<serde_json::Value> {
+    use shell_automaton::mempool::mempool_actions::ConsensusOperationMatcher;
+    let matcher = match (block_hash, payload_hash, level, round) {
+        (Some(block_hash), None, None, None) => {
+            ConsensusOperationMatcher::endorsement_branch(block_hash)
+        }
+        //        (None, Some(payload_hash), None, None) => ConsensusOperationMatcher::endorsement_payload(payload_hash),
```
Comment (tizoc) [inline rpc/src/services/dev_services.rs:948]: what about this commented out case? not needed?

File: rpc/src/services/dev_services.rs [hunk 2]
```diff
@@ -910,16 +914,79 @@ pub(crate) async fn get_shell_automaton_endorsing_rights(
...
+pub(crate) async fn get_shell_automaton_preendorsements_status(
+    payload_hash: Option<BlockPayloadHash>,
+    level: Option<i32>,
+    round: Option<i32>,
+    base_time: Option<u64>,
+    env: &RpcServiceEnvironment,
+) -> anyhow::Result<serde_json::Value> {
+    use shell_automaton::mempool::mempool_actions::ConsensusOperationMatcher;
+    let matcher = match (payload_hash, level, round) {
+        //        (Some(payload_hash), None, None) => ConsensusOperationMatcher::endorsement_payload(payload_hash),
```
Comment (tizoc) [inline rpc/src/services/dev_services.rs:975]: Another commented out case

File: rpc/src/services/dev_services.rs (hunk 1 above)
Comment (akoptelov) [reply rpc/src/services/dev_services.rs:948]: Now it is not needed, a stub for the future.

File: rpc/src/services/dev_services.rs (hunk 2 above)
Comment (akoptelov) [reply rpc/src/services/dev_services.rs:975]: The same as above.

**Pull request merged**